### PR TITLE
feat: Add static-site to Python examples

### PR DESCRIPTION
**Pull request by pooyavahidi**
Added static site example using S3, CloudFront, Certificate Manager and Route53.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

**Comment by ryparker** [comment]
@pooyavahidi Could you address the build failure:

```sh
Traceback (most recent call last):
  File "app.py", line 20, in <module>
    "CDK_DEPLOY_ACCOUNT", os.environ["CDK_DEFAULT_ACCOUNT"]
  File "/usr/lib64/python3.7/os.py", line 681, in __getitem__
    raise KeyError(key) from None
KeyError: 'CDK_DEFAULT_ACCOUNT'
Subprocess exited with error 1
Error: Process completed with exit code 1.
```

**Comment by pooyavahidi** [comment]
> @pooyavahidi Could you address the build failure:
> 
> ```shell
> Traceback (most recent call last):
>   File "app.py", line 20, in <module>
>     "CDK_DEPLOY_ACCOUNT", os.environ["CDK_DEFAULT_ACCOUNT"]
>   File "/usr/lib64/python3.7/os.py", line 681, in __getitem__
>     raise KeyError(key) from None
> KeyError: 'CDK_DEFAULT_ACCOUNT'
> Subprocess exited with error 1
> Error: Process completed with exit code 1.
> ```

@ryparker This has been fixed now. Could you please review.
Thanks

**Comment by lvthillo** [comment]
@pooyavahidi don't forget that your Certificate form ACM should be created in us-east-1 for CloudFront distributions. So if this stack is created in a different region it won't work with the current setup. You'll probably need something like below and cdk boostrap

```
    def __create_certificate(self, hosted_zone):
        self.certificate = acm.DnsValidatedCertificate(self, 'CrossRegionCertificate',
            domain_name=self.__domain_name,
            hosted_zone= self.hosted_zone,
            region='us-east-1',
        )

```

**Comment by pooyavahidi** [comment]
> @pooyavahidi don't forget that your Certificate form ACM should be created in us-east-1 for CloudFront distributions. So if this stack is created in a different region it won't work with the current setup. You'll probably need something like below and cdk boostrap
> 
> ```
>     def __create_certificate(self, hosted_zone):
>         self.certificate = acm.DnsValidatedCertificate(self, 'CrossRegionCertificate',
>             domain_name=self.__domain_name,
>             hosted_zone= self.hosted_zone,
>             region='us-east-1',
>         )
> ```

@lvthillo thanks for the tip. It's fixed now. 
I have a condition in the `__create_certificate` function to either create a new cert (which is fixed now) or accept the arn of an existing certificate. I added a note in the README to call out if an existing certificate has been provided it should be in the `us-east-1`.

File: python/static-site/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+hosting a static web site.
+
+## Prerequisites
+- A registered domain in Route53.
+- If using a public S3 bucket, a secret phrase needs to be created in the parameter store. The name of the parameter to be passed in the context as `origin_custom_header_parameter_name`. Make sure to create the parameter in the same region as the static site. This parameter will be set in both CloudFront and S3 Bucket Policy to make sure only traffic from CloudFront can access the S3 objects.
+- If you're using an existing certificate, make sure it's in the `us-east-1` region otherwise the deployment fails. See this https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cnames-and-https-requirements.html#https-requirements-aws-region
+- The AWS environment needs to be bootstrapped.
+```sh
+cdk bootstrap aws://123456789012/us-east-1
+```
```
Comment (ryparker) [inline python/static-site/README.md:12]: Would you mind adding the standard python setup/build/deploy/cleanup instructions? You should be able to find those in other python example READMEs.

**Review by ryparker — CHANGES_REQUESTED** [review]

File: python/static-site/README.md (hunk 1 above)
Comment (pooyavahidi) [reply python/static-site/README.md:25]: Done! Please review.

**Review by ryparker — APPROVED** [review]
Thanks Pooya

**Pull request merged**